src/umbraco.cms/businesslogic/Packager/data.cs

```diff
@@ -87,301 +94,317 @@ public static XmlNode GetFromId(int Id, string dataSource, bool reload)
+                retVal.DataTypes = new List<string>(SafeNodeValue(n.SelectSingleNode("datatypes")).Trim(',').Split(','));
+
+                XmlNodeList xmlFiles = n.SelectNodes("files/file");
+                retVal.Files = new List<string>();
+
+                for (int i = 0; i < xmlFiles.Count; i++)
+                    retVal.Files.Add(xmlFiles[i].InnerText);
+
+                retVal.LoadControl = SafeNodeValue(n.SelectSingleNode("loadcontrol"));
+            }
+
             return retVal;
         }
-        
-        public static void Delete(int Id, string dataSource) 
-		{
+
+        public static void Delete(int Id, string dataSource)
+        {
             Reload(dataSource);
-			// Remove physical xml file if any
+            // Remove physical xml file if any
             //PackageInstance p = new PackageInstance(Id);
 
             //TODO DELETE PACKAGE FOLDER... 
             //p.Folder
 
-			XmlNode n = data.GetFromId(Id, dataSource, true);
+            XmlNode n = data.GetFromId(Id, dataSource, true);
             if (n != null)
             {
                 data.Source.SelectSingleNode("/packages").RemoveChild(n);
-                data.Source.Save(dataSource);    
+                data.Source.Save(dataSource);
             }
-			
-		}
-
-		public static void UpdateValue(XmlNode n, string Value) 
-		{
-			if (n.FirstChild != null)
-				n.FirstChild.Value = Value;
-			else 
-			{
-                n.AppendChild(Source.CreateTextNode(Value));
-			}
-			//Save();
-		}
-
-		public static void Save(PackageInstance package, string dataSource) 
-		{
-			try
-			{
-                Reload(dataSource);
-                XmlNode _xmlDef = GetFromId(package.Id, dataSource, false);
-                _xmlDef.Attributes.GetNamedItem("name").Value = package.Name;
-                _xmlDef.Attributes.GetNamedItem("version").Value = package.Version;
-                _xmlDef.Attributes.GetNamedItem("url").Value = package.Url;
-                _xmlDef.Attributes.GetNamedItem("packagepath").Value = package.PackagePath;
-                _xmlDef.Attributes.GetNamedItem("repositoryGuid").Value = package.RepositoryGuid;
-                _xmlDef.Attributes.GetNamedItem("packageGuid").Value = package.PackageGuid;
 
-                _xmlDef.Attributes.GetNamedItem("hasUpdate").Value = package.HasUpdate.ToString();
-                _xmlDef.Attributes.GetNamedItem("enableSkins").Value = package.EnableSkins.ToString();
-                _xmlDef.Attributes.GetNamedItem("skinRepoGuid").Value = package.SkinRepoGuid.ToString();
+        }
 
-                    
+        public static void UpdateValue(XmlNode n, string Value)
+        {
+            if (n.FirstChild != null)
+                n.FirstChild.Value = Value;
+            else
+            {
+                n.AppendChild(Source.CreateTextNode(Value));
+            }
+            //Save();

```

Comment:
Turns out the method `UpdateValue` doesn't actually get called ever, I'll review why it was there